Prompts

genagain opened the pull request:
In this PR, I've  implemented rendering welcome text to address #34 . In addition, I've added prompts to every static command. This way the prompt can be decoupled from the output because the data structure for the output isn't the same in every case. Finally, the clear command clears the header element as well as the commands.

Commit:
created render welcome prompt method

Commit:
rendering formatted welcome ascii art

Commit:
decreased font size for welcome ascii art

Commit:
refactored render method to render templates in the header

Commit:
added new lines in welcome text

Commit:
reformatted help and yes commands and prompts projects, help and yes

Commit:
added prompt key value pair to each command that needs a prompt

Commit:
deleted console.log for debugging

Commit:
added a &nbsp; for the hello command

Commit:
extended clear command to clear the header element as well

Commit:
created welcome template

mikehwagz commented:
Dope! I really like how you implemented the prompts and how you refactored `render`. My only suggestion before we merge would be around your approach for the welcome messaging. The only thing that is dynamic is the date & time part. Thus, all static content such as the ascii and welcome prompt can just be inserted directly into the header in `index.html`.  Then, you can revert the datetime logic back to how we had it before. Only difference would be instead of inserting directly into the header, insert it directly into something like `<p class="js-datetime"></p>` before the ascii and welcome prompt.

Commit:
added js-datetime js class and static content to header element

Commit:
append current datetime to datetime element

Commit:
removed welcome method and template

Commit:
moved js-datetime class before the static welcome text

genagain commented:
Thank you for the feedback! I'm glad to hear that you like how I implemented the prompts and refactored the render method. I definitely see where you are coming from about decoupling the static content and the dynamic content for the welcome message. After I implemented your suggestion, I noticed that the current datetime shifts the static content down as the page is loading. In the previous implementation, all of the welcome text was rendered in one fell swoop because the welcome template was being appended to the header. It may be worth comparing the current implementation to the previous implementation. To checkout that commit on this branch and compare you can run the following commands.

```
git checkout prompts
git checkout 7d174850bf35ea26c62b5811413c337d094254bc
gulp
```

mikehwagz commented:
I imagined that we would hide everything else during the preloader, and only reveal the header and cursor after the preloader is finished. This can be achieved easily with CSS. With this approach, the content jump you've described is a non-issue. So I think we should stick to this approach.
